src/linear_algebra/quadratic_form.lean

```diff
@@ -52,26 +52,36 @@ quadratic form, homogeneous polynomial, quadratic polynomial
 
 universes u v w
 variables {R : Type u} {M : Type v} [add_comm_group M] [ring R]
-variables {R₁ : Type u} [comm_ring R₁] [module R₁ M]
+variables {R₁ : Type u} [comm_ring R₁]
 
+namespace quadratic_form
 /-- Up to a factor 2, `Q.polar` is the associated bilinear form for a quadratic form `Q`.d
 
 Source of this name: https://en.wikipedia.org/wiki/Quadratic_form#Generalization
 -/
-def quadratic_form.polar (f : M → R) (x y : M) :=
+def polar (f : M → R) (x y : M) :=
 f (x + y) - f x - f y
 
-variables [module R M]
+lemma polar_add (f g : M → R) (x y : M) :
+  polar (f + g) x y = polar f x y + polar g x y :=
+by { simp only [polar, pi.add_apply], abel }
+
+lemma polar_comm (f : M → R₁) (x y : M) : polar f x y = polar f y x :=
+by rw [polar, polar, add_comm, sub_sub, sub_sub, add_comm (f x) (f y)]
+
+end quadratic_form
+
+variables [module R M] [module R₁ M]
 
 open quadratic_form
 /-- A quadratic form over a module. -/
 structure quadratic_form (R : Type u) (M : Type v) [ring R] [add_comm_group M] [module R M] :=
 (to_fun : M → R)
 (to_fun_smul : ∀ (a : R) (x : M), to_fun (a • x) = a * a * to_fun x)
-(polar_add_left : ∀ (x x' y : M), polar to_fun (x + x') y = polar to_fun x y + polar to_fun x' y)
-(polar_smul_left : ∀ (a : R) (x y : M), polar to_fun (a • x) y = a • polar to_fun x y)
-(polar_add_right : ∀ (x y y' : M), polar to_fun x (y + y') = polar to_fun x y + polar to_fun x y')
-(polar_smul_right : ∀ (a : R) (x y : M), polar to_fun x (a • y) = a • polar to_fun x y)
+(polar_add_left' : ∀ (x x' y : M), polar to_fun (x + x') y = polar to_fun x y + polar to_fun x' y)
```

Comment:
I think another option would be to make the structure fields protected with the new `@[protect_proj]` attribute on this structure. And then add the simp lemmas to the root namespace. (I don't expect future name clashes with `polar_add_left` in the root namespace.)
But the current approach is a bit more conservative, which is also fine.

Reply:
That's also a nice solution. I think I prefer to keep everything in the `quadratic_form` namespace, just like in the other `..._form` definitions. Do we have guidelines on how to structure namespaces?